chat-plugins/panagrams-beta.js

```diff
@@ -152,4 +152,135 @@ exports.commands = {
 					"The type of this Pokemon is: <b>" + Tools.getTemplate(room.panagram.answer).types + "</b>";
 		this.sendReplyBox("Panagram hint: <br />The Pokemon is <b>" + answer_length + "</b> characters long.  <br />" + hint);
 	}
-};
\ No newline at end of file
+};*/
+
+//Panagrams by SilverTactic (Siiilver) and panpawn
+//#Tyme4BugHunting
+var pGames = exports.pGames = {};
+
+function mix(word) {
+	var arr = [];
+	for (var k = 0; k < word.length; k++) {
+		arr.push(word[k]);
+	}
+	var a, b, i = arr.length;
+	while (i) {
+		a = Math.floor(Math.random() * i);
+		i--;
+		b = arr[i];
+		arr[i] = arr[a];
+		arr[a] = b;
+	}
+	return arr.join('').toString();
+}
+
+var Panagram = (function () {
+	function Panagram(room, category) {
+		this.room = room;
+		this.category = category;
+		switch (toId(category)) {
+		case 'items':
+		case 'item':
+			this.dex = Tools.data.Items;
+			break;
+		case 'abilities':
+		case 'ability':
+			this.dex = Tools.data.Abilities;
+			break;
+		case 'moves':
+		case 'move':
+			this.dex = Tools.data.Movedex;
+			break;
+		default:
+			this.dex = Tools.data.Pokedex;
+		}
+		do {
+			this.answer = this.dex[Object.keys(this.dex)[Math.floor(Math.random() * Object.keys(this.dex).length)]];
+		} while (this.answer.num < 1 || this.answer.baseSpecies);
+		this.mixed = mix(this.category === 'pokemon' ? toId(this.answer.species) : this.answer.name.toLowerCase());
+
+		this.room.add('|html|<div class = "infobox"><center>A game of Panagram was started! Scrambled word: <b>' + this.mixed + '</b>. Category: <b>' + category + '</b><br>' +
+			'<small>Enter your answer into the chat to guess!</small></center>'
+		);
+		this.users = {};
+		this.guessed = {};
+
+		var Game = this;
+		this.room.chat = function (user, message, connection) {
+			message = CommandParser.parse(message, this, user, connection);
+			if (Game.dex[toId(message)]) {
+				var guess = Game.dex[toId(message)];
+				var guessName = guess.id || toId(guess.species);
+				if (Game.users[user.userid]) return user.sendTo(this, 'You cannot guess more than once per game of panagram.');
+				for (var i in Game.users)
+					if (Game.users[i].latestIp === user.latestIp) return user.sendTo(this, 'One of your alts have already guessed once in this game of panagram.');
```

Comment:
I don't think we need to check this.  I don't really want a limit to the number of guesses a user can do /o/